### ISSUE TITLE
Set Font in PdfGraphics2D not working / PdfGraphics2D manipulates set Font (not expected)

### ISSUE THREAD

**Issue by mppperez** [LibrePDF/OpenPDF#587]
**Describe the bug**
We didn't update for a while but the changes from 2019 (commit 15e89ca38126ae1dbe5ec869708370321f0ed265) break all generated PDFs (wrong font used).

This can be reproduced with an existing test : `com.lowagie.text.pdf.fonts.FontTest#testBoldSimulationAndStrokeWidth`. 

Executing the test on my machine (Ubuntu, Java 11) let the `compositeFontDrawer` in `PdfGraphics2D` choose the font "Deja Vu Bold" instead of "Helvetica Bold". Note: since the `DefaultFontMapper` from OpenPDF is used it will always fallback to Helvetica anyways (`logicalName` is Deja Vu...): https://github.com/mppperez/OpenPDF/blob/f773081d4f169c6d84579a855ebbed1e93cfd8e9/openpdf/src/main/java/com/lowagie/text/pdf/DefaultFontMapper.java#L123-L193

So this won't have direct affect in the test but if a custom mapper is used that is not restricting to only those 3 fonts you may encounter a problem. _It is not expected that the PdfGraphics2D is changing the specified font on it's own although it was correctly used_. It looks like a project specific issue was built into OpenPDF (see below).

**To Reproduce**
Execute above mentioned test on a machine without Helvetica installed.

**Expected behavior**
Use the provided font (backwards compatibility is missing) in OpenPDF.

**Screenshots**
![Screenshot_562](https://user-images.githubusercontent.com/44845832/129175724-6ca3daee-d024-46f5-a4df-466b1c780d75.png)

**System (please complete the following information):**
 - OS: Ubuntu 20.04
 - Used Font: Helvetica

**Update**
Unfortunately I cannot reconstruct why this fix (see commit above) for Windows was built... or initially Windows and then changed to Mac (see commit history).  It looks like a project specific issue that was implemented in OpenPDF rather than in the appropriate project. At least making the FontDrawer a configurable property seems to be the more proper solution or just using the library with correct fonts instead of logical ones that may not be found.

Furthermore `PdfGraphics2D.CompositeFontDrawer#splitStringIntoDisplayableParts` will even use different fonts just because they are in the same family as the specified one. If the user of the library wants to write text to a pdf and specifies a font that is not containing (some) characters specified in the text then this should be handled on the users side (may throw an exception? logging and skip? ...) but not in OpenPDF itself?

**Comment by mppperez** [comment]
followup: expected behaviour would be also to fail with an exception that the font were not found rather then just using another one that breaks with policies that documents should have a predefined style just only because it is in the same font family...?

**Comment by mppperez** [comment]
Update: unfortunately I cannot reconstruct why this fix (see commit above) for Windows was built... or initially Windows and then changed to Mac. 

Additionally the `splitStringIntoDisplayableParts`(which was added with the commit) looks like a fix for a project specific problem that were built in core mechanics of OpenPDF? If we want to write a PDF document with PdfGraphics2D then we explicitly set the font for each text we're writing. E.g. If we have arabic on the left side and chinese on the right side of the document then we also set the necessary font for it. With the current code the library checks for composite fonts (unfortunately also with a lot of reflection...) if the character can be written by another font in the family and if the first one already can write all characters it is simply choosing that font regardless of the other fonts (and specifically the correct specified one that is maybe later in that list?).

I will try working on a fix and create a pull request... with more backgrounds this could be ofc better. With my current understandings I would simply refactor the stuff that were added into an own handler that is set by default (backwards compatibility) but users of the library (or especially PdfGraphics2D) can deactivate this additional font resolving mechanism  by removing the handler / resolver or extend / customize it by adding own handlers/resolvers.
